### ISSUE TITLE
Podcasts - sorting by episode tag is broken

### ISSUE THREAD

**Issue by Bougakov** [jaedb/Iris#576]
**Describe the bug**

Some podcasts turn out to have only a `<itunes:episode>n</itunes:episode>` and `<itunes:type>episodic</itunes:type>`, but not the date in their RSS feeds. Mopidy-Iris displays those in a random order.

`<channel>` -> `<title>` display is broken as well.

I've initially filed this bug in https://github.com/tkem/mopidy-podcast/issues/57 - but turns out `Mopidy-Podcast` is not at fault, but `Mopidy-Iris` is.

**To Reproduce**
1. Add a sample podcast, for example https://feeds.megaphone.fm/WWO6655869236
2. Check manually the value of `<itunes:episode>` tags
3. Open Podcasts section in Iris, see that they appear in random order.

**Expected behavior**
Correct order of episodes.

**Screenshots**

Correct sorting: 

 ![image](https://user-images.githubusercontent.com/2833077/88521799-bfcc5d00-cff5-11ea-8425-6a29d91ca507.png)

Incorrect sorting:

> ![image](https://user-images.githubusercontent.com/1763243/88490170-50794d80-cfa2-11ea-9b4a-dd08db0d79d0.png)

**Environment details:**
 - OS: raspbian
 - Browser: MS Edge (Chromium)
 - Iris version: latest from pip

**Comment by adigandhi** [comment]
Yeah, I see all podcasts arranged in a very weird way, and latest one's seem towards the end..

**Comment by jaedb** [comment]
This is (a more thorough) duplicate of #550
